ext/KrylovKitChainRulesCoreExt/eigsolve.jl

```diff
@@ -51,7 +51,7 @@ function make_eigsolve_pullback(config, f, fᴴ, x₀, howmany, which, alg_prima
         n_vals = isnothing(_n_vals) ? 0 : _n_vals
         n_vecs = isnothing(_n_vecs) ? 0 : _n_vecs
         n = max(n_vals, n_vecs)
-        if n < length(vals) && vals[n + 1] == conj(vals[n])
+        if n != 0 && n < length(vals) && vals[n + 1] == conj(vals[n])
```

Comment:
Why does the `n==0` case need to be excluded? Does it happen in a VUMPS context that one of the eigenvalue problems has two complex conjugated eigenvalues of largest magnitude?

Reply:
Yes, there are always degenerate eigenvalues when initializing a large unit cell PEPS randomly.

Reply:
This might be an artifact of the way you supply your operators: if I understood correctly you always transfer over a single site, and take all the collection of all tensors as the "eigenvector", which indeed trivially has eigenvalues with phases `2 * pi * k / L` because you can cycle through the unitcell without these affecting the final result, leading to conjugate eigenvalues whenever the unitcell is even(?).

I don't know the answer to this: is it in this case fair to just consider the one you obtain first? It seems like this might be sensitive to initial conditions etc, and I am not sure if this rrule (mathematically) is correct.

Reply:
I clearly misread this, the `n==0` case just leads to the indexing error with `vals[n]` and does definitely need to be excluded. May I suggest to use simply

`if 0 < n < length(vals) && vals[n + 1] == conj(vals[n])`

Also, using the exact equality as a way to "hope" that this will only be satisfied in the case where the original `schursolve` was performed in real arithmetic is perhaps a bit of a hack. However, at the level of `eigsolve`, I don't think it can still be detected whether the original problem represented a real-value linear operator or a complex-valued one, so I don't have a better suggestion for that.